frontend/src/store/useUserStore.ts

```diff
@@ -0,0 +1,42 @@
+import { create } from 'zustand';
+import { persist } from 'zustand/middleware';
+
+import { User } from '@/types/user';
+
+interface UserState {
+  user: User;
+  setUser: (user: User) => void;
+  reset: () => void;
+}
+
+const initialUser: User = {
+  userId: 0,
+  userName: '',
+  userEmail: '',
+  createdAt: '',
+  userType: 'GUEST',
+};
+
+const useUserStore = create<UserState>()(
+  persist(
+    set => ({
+      user: { ...initialUser },
+
+      setUser: (newUser: User) => {
+        set(() => ({ user: { ...newUser } }));
+      },
```

Comment:
```suggestion
      setUser: (newUser) => {
        set({ user: { ...newUser } });
      },
```
불필요한 화살표들을 제거할 수 있을 것 같아요 !
아래에도 공통입니다